From 94cfff69993553184c912b74456c643a1083454e Mon Sep 17 00:00:00 2001
From: Arturo Rinaldi <arty.net2@gmail.com>
Date: Fri, 11 Mar 2016 12:31:25 +0100
Subject: [PATCH] adding missing functions for linino-webpanel/Arduino-OS

---
 modules/luci-base/luasrc/sys.lua | 46 ++++++++++++++++++++++++++++++++++++++++
 1 file changed, 46 insertions(+)

diff --git a/modules/luci-base/luasrc/sys.lua b/modules/luci-base/luasrc/sys.lua
index a972717..5ab9d5b 100644
--- a/modules/luci-base/luasrc/sys.lua
+++ b/modules/luci-base/luasrc/sys.lua
@@ -97,6 +97,52 @@ function reboot()
 	return os.execute("reboot >/dev/null 2>&1")
 end
 
+--- Returns the system type, cpu name and installed physical memory.
+-- @return      String containing the system or platform identifier
+-- @return      String containing hardware model information
+-- @return      String containing the total memory amount in kB
+-- @return      String containing the memory used for caching in kB
+-- @return      String containing the memory used for buffering in kB
+-- @return      String containing the free memory amount in kB
+-- @return      String containing the cpu bogomips (number)
+function sysinfo()
+	local cpuinfo = fs.readfile("/proc/cpuinfo")
+	local meminfo = fs.readfile("/proc/meminfo")
+
+	local memtotal = tonumber(meminfo:match("MemTotal:%s*(%d+)"))
+	local memcached = tonumber(meminfo:match("\nCached:%s*(%d+)"))
+	local memfree = tonumber(meminfo:match("MemFree:%s*(%d+)"))
+	local membuffers = tonumber(meminfo:match("Buffers:%s*(%d+)"))
+	local bogomips = tonumber(cpuinfo:match("[Bb]ogo[Mm][Ii][Pp][Ss].-: ([^\n]+)")) or 0
+	local swaptotal = tonumber(meminfo:match("SwapTotal:%s*(%d+)"))
+	local swapcached = tonumber(meminfo:match("SwapCached:%s*(%d+)"))
+	local swapfree = tonumber(meminfo:match("SwapFree:%s*(%d+)"))
+
+	local system =
+			cpuinfo:match("system type\t+: ([^\n]+)") or
+			cpuinfo:match("Processor\t+: ([^\n]+)") or
+			cpuinfo:match("model name\t+: ([^\n]+)")
+
+	local model =
+			luci.util.pcdata(fs.readfile("/tmp/sysinfo/model")) or
+			cpuinfo:match("machine\t+: ([^\n]+)") or
+			cpuinfo:match("Hardware\t+: ([^\n]+)") or
+			luci.util.pcdata(fs.readfile("/proc/diag/model")) or
+			nixio.uname().machine or
+			system
+
+	return system, model, memtotal, memcached, membuffers, memfree, bogomips, swaptotal, swapcached, swapfree
+end
+
+--- Returns the system load average values.
+-- @return      String containing the average load value 1 minute ago
+-- @return      String containing the average load value 5 minutes ago
+-- @return      String containing the average load value 15 minutes ago
+function loadavg()
+    local info = nixio.sysinfo()
+    return info.loads[1], info.loads[2], info.loads[3]
+end
+
 function syslog()
 	return luci.util.exec("logread")
 end
