diff --git a/package/libs/openssl/Makefile b/package/libs/openssl/Makefile
index 3d13fe0..50ccfdb 100644
--- a/package/libs/openssl/Makefile
+++ b/package/libs/openssl/Makefile
@@ -34,6 +34,8 @@ PKG_CONFIG_DEPENDS:= \
 	CONFIG_OPENSSL_WITH_SSL3
 
 include $(INCLUDE_DIR)/package.mk
+include $(INCLUDE_DIR)/host.mk
+include $(INCLUDE_DIR)/host-build.mk
 
 ifneq ($(CONFIG_CCACHE),)
 HOSTCC=$(HOSTCC_NOCACHE)
@@ -131,6 +133,63 @@ endif
 
 STAMP_CONFIGURED := $(STAMP_CONFIGURED)_$(subst $(space),_,$(OPENSSL_OPTIONS))
 
+# HOST Build
+
+
+HOST_CPPFLAGS += $(HOST_FPIC)
+
+ifeq ($(HOST_ARCH),x86_64)
+  OPENSSL_HOST_TARGET:=linux-x86_64
+else
+  OPENSSL_HOST_TARGET:=linux-generic32
+endif
+
+define Host/Configure
+	[ -f $(HOST_STAMP_CONFIGURED) ] || { \
+		rm -f $(HOST_BUILD_DIR)/*.so.* $(HOST_BUILD_DIR)/*.a; \
+		find $(HOST_BUILD_DIR) -name \*.o | xargs rm -f; \
+	}
+	(cd $(HOST_BUILD_DIR); \
+		./Configure "$(OPENSSL_HOST_TARGET)" \
+			--prefix=/usr \
+			--openssldir=/etc/ssl \
+			$(HOST_CPPFLAGS) \
+			$(HOST_LDFLAGS) -ldl \
+			$(OPENSSL_NO_CIPHERS) \
+			shared zlib-dynamic \
+	)
+	# XXX: OpenSSL "make depend" will look for installed headers before its own,
+	# so remove installed stuff first
+	-$(SUBMAKE) -j1 clean-staging
+	+$(MAKE) $(PKG_JOBS) -C $(HOST_BUILD_DIR) \
+		MAKEDEPPROG="$(HOSTCC)" \
+		depend
+endef
+
+define Host/Compile
+	+$(MAKE) $(PKG_JOBS) -C $(HOST_BUILD_DIR) \
+		CC="$(HOSTCC)" \
+		all
+	+$(MAKE) $(PKG_JOBS) -C $(HOST_BUILD_DIR) \
+		CC="$(HOSTCC)" \
+		build-shared
+	# Work around openssl build bug to link libssl.so with libcrypto.so.
+	-rm $(HOST_BUILD_DIR)/libssl.so.*.*.*
+	+$(MAKE) $(PKG_JOBS) -C $(HOST_BUILD_DIR) \
+		CC="$(HOSTCC)" \
+		do_linux-shared
+endef
+
+define Host/Install
+	$(MAKE) -C $(HOST_BUILD_DIR) \
+		CC="$(HOSTCC)" \
+		INSTALL_PREFIX="$(STAGING_DIR_HOST)" \
+		install
+endef
+
+##########
+
+
 define Build/Configure
 	[ -f $(STAMP_CONFIGURED) ] || { \
 		rm -f $(PKG_BUILD_DIR)/*.so.* $(PKG_BUILD_DIR)/*.a; \
@@ -217,3 +276,4 @@ endef
 
 $(eval $(call BuildPackage,libopenssl))
 $(eval $(call BuildPackage,openssl-util))
+$(eval $(call HostBuild))
