From ea21c1bd1cfab2ded37fe141b34c7bf14313c133 Mon Sep 17 00:00:00 2001
From: m13 <mimmo@linino.org>
Date: Fri, 23 May 2014 00:36:57 +0200
Subject: [PATCH 02/24] uboot: Add linino_chow_config

---
 Makefile | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/Makefile b/Makefile
index 56f2e6c..3fda46f 100644
--- a/Makefile
+++ b/Makefile
@@ -343,6 +343,18 @@ ifdef BOARD_STRING
 endif
 	@./mkconfig -a ap121 mips mips ap121 ar7240 ar7240 linino
 	
+linino_chow_config		    : 	unconfig
+	@echo '#define CONFIG_AP123 1' >>include/config.h
+	@echo '#define CONFIG_ROOTFS_TYPE "rootfstype=squashfs"' >>include/config.h
+	@echo "#define CONFIG_AR7240 1" >>include/config.h
+	@echo "#define CONFIG_WASP 1" >>include/config.h
+	@echo '#define CFG_AG7240_NMACS 1' >>include/config.h
+	@echo "#define FLASH_SIZE $(FLASH_SIZE)" >>include/config.h
+ifdef BOARD_STRING
+	@echo "#define BOARD_STRING 1" >>include/config.h
+endif
+	@./mkconfig -a ap123 mips mips ap123 ar7240 ar7240 linino
+	
 #########################################################################
 #########################################################################
 #########################################################################
-- 
2.10.1

