From 059530ecdbb47e12f97f3ddcd552d0498f90fed6 Mon Sep 17 00:00:00 2001
From: m13 <mimmo@linino.org>
Date: Wed, 18 Mar 2015 17:06:13 +0100
Subject: [PATCH 16/24] setting default baudrate

---
 include/configs/ap123.h | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/include/configs/ap123.h b/include/configs/ap123.h
index 442ec80..3e84cb0 100644
--- a/include/configs/ap123.h
+++ b/include/configs/ap123.h
@@ -155,10 +155,10 @@
 #		if CONFIG_LININO_IO
 #			define CONSOLE_LININOIO_DEFAULT	"console=spicons\0"
 #		else
-#			define CONSOLE_DEFAULT		"console=ttyS0,115200\0"
+#			define CONSOLE_DEFAULT		"console=ttyATH0,460800\0"
 #		endif
 #		undef CONFIG_BAUDRATE
-#		define CONFIG_BAUDRATE	115200
+#		define CONFIG_BAUDRATE	250000
 #	else
 #		define ATH_U_FILE	u-boot.bin
 #		define ATH_F_FILE	${bc}-jffs2
@@ -220,7 +220,7 @@
 #ifdef CONFIG_LININO
 #define	CONFIG_BOOTARGS
 #else
-#define	CONFIG_BOOTARGS		"console=ttyS0,115200 root=" ATH_ROOT_DEV " rootfstype=squashfs init=/sbin/init " MTDPARTS_DEFAULT
+#define	CONFIG_BOOTARGS		"console=ttyATH0,460800 root=" ATH_ROOT_DEV " rootfstype=squashfs init=/sbin/init " MTDPARTS_DEFAULT
 #endif
 
 #undef CFG_PLL_FREQ
-- 
2.10.1

