From 64dc18afda7b40c467139a28c82fb8f019c3d7c0 Mon Sep 17 00:00:00 2001
From: Aurelio Colosimo <aurelio@aureliocolosimo.it>
Date: Sun, 22 Nov 2015 10:33:25 +0100
Subject: [PATCH 60/83] MCUIO: lucky shield, enable pullup in pca9555 interrupt
 pin

Signed-off-by: Aurelio Colosimo <aurelio@aureliocolosimo.it>
---
 drivers/mcuio/mcuio-lucky-shield.c | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/drivers/mcuio/mcuio-lucky-shield.c b/drivers/mcuio/mcuio-lucky-shield.c
index 0e9aea7..a05b700 100644
--- a/drivers/mcuio/mcuio-lucky-shield.c
+++ b/drivers/mcuio/mcuio-lucky-shield.c
@@ -116,6 +116,15 @@ static int mcuio_lucky_probe(struct mcuio_device *mdev)
 				return -EINVAL;
 			}
 
+			/* HACK this is needed to enable pullup */
+			ret = devm_gpio_request_one(&mdev->dev, gpio,
+				GPIOF_DIR_IN, "lucky-shield");
+			if (ret < 0)
+				return ret;
+			gpio_direction_output(gpio, 1);
+			gpio_direction_input(gpio);
+			devm_gpio_free(&mdev->dev, gpio);
+
 			i->info.irq = gpio_to_irq(gpio);
 		}
 
-- 
2.10.1

